### PR TITLE
authz: update troubleshooting guide for authZ policy

### PR DESCRIPTION
**Pull request by yangminzhu**

### CODE REVIEW

**Comment by yangminzhu** [comment]
@adammil2000 @rcaballeromx @liminw Would you mind to take a look at this? The current content is still using alpha RBAC policy is super out dated at this moment. Thanks.

**Comment by yangminzhu** [comment]
@liminw @adammil2000 @rcaballeromx ping

File: content/en/docs/ops/common-problems/security-issues/index.md [hunk 1]
```diff
@@ -84,17 +84,11 @@ matching requests should flow through. If all requests continue to be denied, yo
...
+1. Make sure that your authorization policies are in the right same namespace (by checking "metadata"/”namespace” line).
 
-1. Make sure that your service role and service role binding policies don't use any HTTP only fields
-for TCP services. Otherwise, Istio ignores the policies as if they didn't exist.
-
-1. In Kubernetes environment, make sure all services in a `ServiceRole` object are in the same namespace as the
-`ServiceRole` itself. For example, if a service in a `ServiceRole` object is `a.default.svc.cluster.local`, the `ServiceRole` must be in the
-`default` namespace (`metadata/namespace` line should be `default`). For non-Kubernetes environments, all `ServiceRoles` and `ServiceRoleBindings`
-for a mesh should be in the same namespace.
+1. Make sure that your authorization policies don't use any HTTP only fields for TCP traffic. Otherwise, Istio ignores the policies as if they didn't exist.
```
Comment (liminw) [inline content/en/docs/ops/common-problems/security-issues/index.md]: didn't exist =>don't exist. 

It that true for deny policies?

Comment (yangminzhu) [reply content/en/docs/ops/common-problems/security-issues/index.md]: Yeah, it's slightly different here. Updated and added another bullet point for deny policy.

File: content/en/docs/ops/common-problems/security-issues/index.md [hunk 2]
```diff
@@ -84,17 +84,15 @@ matching requests should flow through. If all requests continue to be denied, yo
...
-1. Make sure that your service role and service role binding policies don't use any HTTP only fields
-for TCP services. Otherwise, Istio ignores the policies as if they didn't exist.
+1. Make sure that your authorization policies with ALLOW action don't use any HTTP only fields for TCP traffic.
+Otherwise, Istio ignores the ALLOW policies as if they don't exist.
 
-1. In Kubernetes environment, make sure all services in a `ServiceRole` object are in the same namespace as the
-`ServiceRole` itself. For example, if a service in a `ServiceRole` object is `a.default.svc.cluster.local`, the `ServiceRole` must be in the
-`default` namespace (`metadata/namespace` line should be `default`). For non-Kubernetes environments, all `ServiceRoles` and `ServiceRoleBindings`
-for a mesh should be in the same namespace.
+1. Make sure that your authorization policies with DENY action don't use any HTTP only fields for TCP traffic.
```
Comment (frankbu) [inline content/en/docs/ops/common-problems/security-issues/index.md]: This and the above entry seem to be a convoluted way of saying that the HTTP only fields are ignored, which is what it already says in https://deploy-preview-6767--preliminary-istio.netlify.com/docs/concepts/security/#using-istio-authorization-on-plain-tcp-protocols. I would suggest just one entry that says something like:
```
Make sure that authorization policies for TCP workloads don't use any HTTP only fields.
HTTP only fields are ignored for TCP traffic and consequently have no effect on the policy.
```

File: content/en/docs/ops/common-problems/security-issues/index.md [hunk 3]
```diff
@@ -84,17 +84,15 @@ matching requests should flow through. If all requests continue to be denied, yo
 
 1. Make sure there is no typo in your policy YAML file.
```
Comment (frankbu) [inline content/en/docs/ops/common-problems/security-issues/index.md]: This seems unhelpful. The yaml parser and analyzer should complain about many typos. I would suggest instead describing specific common typos that can slip through, or remove this item. Just saying "typo" is too general IMO.

File: content/en/docs/ops/common-problems/security-issues/index.md [hunk 4]
```diff
@@ -84,17 +84,15 @@ matching requests should flow through. If all requests continue to be denied, yo
...
 
-1. Avoid enabling authorization for Istio Control Planes Components, including Mixer, Pilot, Ingress. Istio authorization policy is designed for authorizing access to services in Istio Mesh. Enabling it for Istio Control Planes Components may cause unexpected behavior.
+1. Avoid enabling authorization for istiod. Istio authorization policy is designed for authorizing access to workloads in Istio Mesh. Enabling it for istiod may cause unexpected behavior.
 
-1. Make sure that your `ServiceRoleBinding` and referred `ServiceRole` objects are in the same namespace (by checking "metadata"/”namespace” line).
+1. Make sure that your authorization policies are in the right namespace (by checking "metadata"/”namespace” line).
```
Comment (frankbu) [inline content/en/docs/ops/common-problems/security-issues/index.md]:
```suggestion
1. Make sure that your authorization policies are in the right namespace (as specified in `metadata/namespace` field).
```

File: content/en/docs/ops/common-problems/security-issues/index.md [hunk 5]
```diff
@@ -139,47 +133,47 @@ you ensure Pilot is working as expected:
...
+    2020-03-05T23:43:21.623645Z   debug   rbac   added TCP filter to filter chain 0
+    2020-03-05T23:43:21.623648Z   debug   rbac   added TCP filter to filter chain 1
     {{< /text >}}
 
-    It means Pilot generated:
+    It means istiod generated:
 
-    - An empty config for `sleep.foo.svc.cluster.local` as there is no authorization policies matched
-      and Istio denies all requests sent to this service by default.
+    - A HTTP filter config with policy `ns[foo]-policy[ext-authz-server]-rule[0]` for workload with labels `app=ext-authz-server,...`.
```
Comment (frankbu) [inline content/en/docs/ops/common-problems/security-issues/index.md]:
```suggestion
    - An HTTP filter config with policy `ns[foo]-policy[ext-authz-server]-rule[0]` for workload with labels `app=ext-authz-server,...`.
```

File: content/en/docs/ops/common-problems/security-issues/index.md [hunk 6]
```diff
@@ -139,47 +133,47 @@ you ensure Pilot is working as expected:
...
+    2020-03-05T23:43:21.623546Z   debug   rbac   building v1beta1 policy
+    2020-03-05T23:43:21.623572Z   debug   rbac   constructed internal model: &{Permissions:[{Services:[] Hosts:[] NotHosts:[] Paths:[] NotPaths:[] Methods:[] NotMethods:[] Ports:[] NotPorts:[] Constraints:[] AllowAll:true v1beta1:true}] Principals:[{Users:[] Names:[cluster.local/ns/istio-system/sa/istio-ingressgateway-service-account] NotNames:[] Group: Groups:[] NotGroups:[] Namespaces:[] NotNamespaces:[] IPs:[] NotIPs:[] RequestPrincipals:[] NotRequestPrincipals:[] Properties:[] AllowAll:false v1beta1:true}]}
+    2020-03-05T23:43:21.623625Z   debug   rbac   generated policy ns[foo]-policy[ext-authz-server]-rule[0]: permissions:<and_rules:<rules:<any:true > > > principals:<and_ids:<ids:<or_ids:<ids:<authenticated:<principal_name:<exact:"spiffe://cluster.local/ns/istio-system/sa/istio-ingressgateway-service-account" > > > > > > >
+    2020-03-05T23:43:21.623645Z   debug   rbac   added TCP filter to filter chain 0
+    2020-03-05T23:43:21.623648Z   debug   rbac   added TCP filter to filter chain 1
     {{< /text >}}
 
-    It means Pilot generated:
+    It means istiod generated:
```
Comment (frankbu) [inline content/en/docs/ops/common-problems/security-issues/index.md]:
```suggestion
    This shows that Istiod generated:
```

File: content/en/docs/ops/common-problems/security-issues/index.md [hunk 7]
```diff
@@ -139,47 +133,47 @@ you ensure Pilot is working as expected:
...
-    $ kubectl logs $(kubectl -n istio-system get pods -l istio=pilot -o jsonpath='{.items[0].metadata.name}') -c discovery -n istio-system | grep rbac
+    $ kubectl logs $(kubectl -n istio-system get pods -l app=istiod -o jsonpath='{.items[0].metadata.name}') -c discovery -n istio-system | grep rbac
     {{< /text >}}
 
 1. Check the output and verify:
 
     - There are no errors.
-    - There is a `"built filter config for ..."` message which means the filter is generated
-      for the target service.
+    - There is a `building v1beta1 policy` message which means the filter is generated
```
Comment (frankbu) [inline content/en/docs/ops/common-problems/security-issues/index.md]:
```suggestion
    - There is a `building v1beta1 policy` message which indicates the filter was generated
```

File: content/en/docs/ops/common-problems/security-issues/index.md (hunk 3 above)
Comment (yangminzhu) [reply content/en/docs/ops/common-problems/security-issues/index.md]: I think the yaml parser and the analyzer doesn't catch the typos, for example, if I apply the following policy:
```yaml
apiVersion: security.istio.io/v1beta1
kind: AuthorizationPolicy
metadata:
 name: test
 namespace: foo
spec:
 action: ALLOW
 selector:
   matchLabels:
     app: test
 typo_rules:
 - from:
   - source:
       principals: ["cluster.local/ns/istio-system/sa/istio-ingressgateway-service-account"]
```

the `typo_rules` field will not be caught and the policy will be accepted. The unknown fields will just be ignored when converting to the AuthorizationPolicy protobuf in Pilot.

File: content/en/docs/ops/common-problems/security-issues/index.md [hunk 8]
```diff
@@ -139,47 +133,47 @@ you ensure Pilot is working as expected:
...
 
-    - An empty config for `sleep.foo.svc.cluster.local` as there is no authorization policies matched
-      and Istio denies all requests sent to this service by default.
+    - An HTTP filter config with policy `ns[foo]-policy[ext-authz-server]-rule[0]` for workload with labels `app=ext-authz-server,...`.
 
-    - An config for `productpage.default.svc.cluster.local` and Istio will allow anyone to access it
-      with GET method.
+    - A TCP filter config with policy `ns[foo]-policy[ext-authz-server]-rule[0]` for workload with labels `app=ext-authz-server,...`.
 
 ## Ensure Pilot distributes policies to proxies correctly
```
Comment (liminw) [inline content/en/docs/ops/common-problems/security-issues/index.md]: Pilot => Istiod?

File: content/en/docs/ops/common-problems/security-issues/index.md (hunk 1 above)
Comment (liminw) [reply content/en/docs/ops/common-problems/security-issues/index.md]: Looks good. Thanks.

**Pull request merged**